spv/verifypayment.go

```diff
@@ -8,29 +8,76 @@ import (
 )
 
 // VerifyPayment verifies whether or not the txs supplied via the supplied spv.Envelope are valid
-func (v *verifier) VerifyPayment(ctx context.Context, initialPayment *Envelope) (bool, error) {
+func (v *verifier) VerifyPayment(ctx context.Context, initialPayment *Envelope, opts ...VerifyOpt) (*bt.Tx, error) {
 	if initialPayment == nil {
-		return false, ErrNilInitialPayment
+		return nil, ErrNilInitialPayment
 	}
-
-	// The tip tx is the transaction we're trying to verify, and it should not have a supplied
-	// Merkle Proof.
-	if initialPayment.IsAnchored() {
-		return false, ErrTipTxConfirmed
+	vOpt := v.opts.clone()
+	for _, opt := range opts{
+		opt(vOpt)
+	}
+	// parse initial tx, fail fast if it isn't a valid tx.
+	tx, err:= bt.NewTxFromString(initialPayment.RawTx)
+	if err != nil{
+		return nil, err
 	}
 
-	valid, err := v.verifyTxs(ctx, initialPayment)
-	if err != nil {
-		return false, err
+	// verify tx fees
+	if vOpt.fees{
+		if err := v.verifyFees(initialPayment, tx, vOpt); err != nil{
+			return nil, err
+		}
+	}
+	// if we are validating proofs or scripts carry out the validation.
+	if vOpt.requiresEnvelope() {
+		// The tip tx is the transaction we're trying to verify, and it should not have a supplied
+		// Merkle Proof.
+		if initialPayment.IsAnchored() {
+			return nil, ErrTipTxConfirmed
+		}
+		if err := v.verifyTxs(ctx, initialPayment,vOpt);err != nil {
+			return nil, err
+		}
 	}
+	return tx, nil
+}
 
-	return valid, nil
+// verifyFees takes the initial payment and iterates the immediate parents in order to gather
+// the satoshis used for each input of the initialPayment tx.
+//
+// If there are no parents the method will fail, also, if there are no fees the method will fail.
+func (v *verifier) verifyFees(initialPayment *Envelope, tx *bt.Tx, opts *verifyOptions) error{
+	if len(initialPayment.Parents) == 0{
+		return ErrCannotCalculateFeePaid
+	}
+	if opts.feeQuote == nil{
```

Comment:
Same with this file

Reply:
Ah no, didn't have it enabled on this repo